File: python/rum/handler.py
import json

def handler(event, context):
    return {
        'statusCode': 200,
        'body': json.dumps(event),
    }

def authorizer(event, context):
    print(event)
    if event.get('type') == 'REQUEST':
        token = (event.get('headers') or {}).get('Authorization', '')[7:]  # Skip 'Bearer '
        method_arn = event.get('methodArn')
    else:
        token = event.get('authorizationToken', '')[7:]
        method_arn = event.get('methodArn')
    return {
        'principalId': 'user',
        'policyDocument': {
            'Version': '2012-10-17',
            'Statement': [
                {
                    'Action': 'execute-api:Invoke',
                    'Effect': 'Allow' if token else 'Deny',
                    'Resource': [method_arn or '*'],
                }
            ]
        }
    }

File: node/authorizer/handler.js
exports.handler = async function(event, context) {
  return {
    statusCode: 200,
    body: JSON.stringify({ event, context }),
  };
}

exports.authorizer = async function(event, context) {
  console.log(event);
  let token;
  let methodArn;
  if (event.type === 'REQUEST') {
    token = (event.headers || {}).Authorization ? event.headers.Authorization.slice(7) : ''; // Skip 'Bearer '
    methodArn = event.methodArn;
  } else {
    token = event.authorizationToken ? event.authorizationToken.slice(7) : '';
    methodArn = event.methodArn;
  }
  return {
    principalId: 'user',
    policyDocument: {
      Version: '2012-10-17',
      Statement: [
        {
          Action: 'execute-api:Invoke',
          Effect: token ? 'Allow' : 'Deny',
          Resource: [methodArn || '*'],
        }
      ]
    }
  };
}
